src/Dumplie/Application/Command/Extension.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Command;
+
+use Dumplie\Application\ServiceLocator;
+
+interface Extension
+{
+    /**
+     * @param Command $command
+     * @return bool
+     */
+    public function expands(Command $command) : bool;
+
+    /**
+     * @param Command $command
+     * @param ServiceLocator $serviceLocator
+     */
+    public function pre(Command $command, ServiceLocator $serviceLocator);
```

Comment:
In some cases `pre` and `post` hooks would need to be in same transaction as command handler.